docs/arcaflow/getting-started.md

```diff
@@ -105,142 +189,488 @@ input:
-2023-03-22T11:25:59+01:00       info            Creating container from image example-plugin...
-2023-03-22T11:26:00+01:00       info            Container started.
-2023-03-22T11:26:00+01:00       info            Step greet is now running...
-2023-03-22T11:26:00+01:00       info            Step greet is now executing ATP...
-2023-03-22T11:26:00+01:00       info            Step "greet" has finished with output success.
-message: Hello, Arca Lot
-```
-
-As you can see, the last line of the output is the output data from the workflow.
+```yaml title="utilities plugin UUID step return YAML"
+output_id: success
+output_data:
+  uuid: bb08484f-6263-4317-9162-be2ae846b438
+debug_logs: ''
+```
 
-!!! tip "Did you know?"
-    Arcaflow provides [Mermaid](https://mermaid.js.org/) markdown in the workflow output that allows you to quickly visualize the workflow in a graphic format. You can grab the Mermaid graph you see in the output and put it into [the Mermaid editor](https://mermaid.live/edit#pako:eNpdjz0OwjAMha8SeaY5QAem3gDGLCZxf6TGiRJHCFW9OxYwtEy2v_ee9LyBT4Ggh3FNTz9jEXMfHNf2mArm2Sycmzj-DMsYyTFxOIKuu1ahXO1UiNR6OM6STU00VG1t3lOtJ-u_qNEvgQtEKhGXoCU3x8Y4kJm0CPS6BhqxreLA8a5WbJJuL_bQj7hWukDLAYWGBfWd-KP7Gz-1Wos).
+[Learn more about plugin schemas &raquo;](/arcaflow/plugins/python/data-model/){ .md-button }
 
-    === "Mermaid markdown"
-        ```
-        flowchart TD
-        subgraph input
-        input.name
-        end
-        input.name-->steps.greet
-        steps.greet-->steps.greet.outputs.success
-        steps.greet.outputs.success-->output
-        ```
+## Writing Plugins
+ Of course you may have specific needs and want to author your own plugins. To aid with this, we provide [SDKs](/arcaflow/plugins/) in popular languages. Let's create a simple hello-world plugin using the Python SDK. We'll publish the code here, and you can find the details in the [Python plugin guide](plugins/python/first.md).
 
-    === "Mermaid rendered flowchart"
-        ```mermaid
-        flowchart TD
-        subgraph input
-        input.name
-        end
-        input.name-->steps.greet
-        steps.greet-->steps.greet.outputs.success
-        steps.greet.outputs.success-->output
-        ```
+```python title="plugin.py"
+#!/usr/local/bin/python3
+from dataclasses import dataclass
+import sys
+from arcaflow_plugin_sdk import plugin
 
-!!! tip
-    Further reading: [Running Arcaflow](running/running.md)
+
+@dataclass
+class InputParams:
+    name: str
+    
+    
+@dataclass
+class SuccessOutput:
+    message: str
+
+
+@plugin.step(
+    id="hello-world",
+    name="Hello world!",
+    description="Says hello :)",
+    outputs={"success": SuccessOutput},
+)
+def hello_world(params: InputParams):
+    return "success", SuccessOutput(f"Hello, {params.name}")
+
+
+if __name__ == "__main__":
+    sys.exit(
+        plugin.run(
+            plugin.build_schema(
+                hello_world,
+            )
+        )
+    )
+```
```

Comment:
Is this code under source control somewhere where it gets tested?

Reply:
No, but it's a good point. I can probably add it as something like a build-your-own plugin repo and then use the file embedding feature we have for markdown here.